### PR TITLE
fix kubelet syncLoop busy loop upon close of updates chan

### PR DESCRIPTION
**Pull request by jdef**
The kubelet syncLoopIteration returns if the update chan is closed, but the syncLoop (controller func) will continue to invoke it regardless of the channel status. Busy loop.

This PR allows syncLoop to terminate if/when the update chan closes (and adds a test for such). The mesos kubelet-executor leverages the channel close to cause kubelet.Run() to exit cleanly.

xref https://github.com/mesosphere/kubernetes-mesos/issues/459

/cc @dchen1107

### CODE REVIEW

**Comment by jdef** [comment]
@dchen1107 PTAL

**Comment by jdef** [comment]
rebased to master

**Comment by jdef** [comment]
not sure why shippable failed, says the logs are too large to show?

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -1896,16 +1896,20 @@ func (kl *Kubelet) syncLoop(updates <-chan PodUpdate, handler SyncHandler) {
...
+	ok = true
 	kl.syncLoopMonitor.Store(time.Now())
 	select {
-	case u, ok := <-updates:
-		if !ok {
+	case u, open := <-updates:
+		if !open {
 			glog.Errorf("Update channel is closed. Exiting the sync loop.")
+			ok = false
 			return
```
Comment (dchen1107) [inline pkg/kubelet/kubelet.go]: why not simply return false here?

File: pkg/kubelet/kubelet.go [hunk 2]
```diff
@@ -1928,6 +1932,7 @@ func (kl *Kubelet) syncLoopIteration(updates <-chan PodUpdate, handler SyncHandl
 		handler.HandlePodSyncs(kl.podManager.GetPods())
 	}
 	kl.syncLoopMonitor.Store(time.Now())
+	return
```
Comment (dchen1107) [inline pkg/kubelet/kubelet.go]: return true?

**Comment by dchen1107** [comment]
LGTM besides nits.

**Comment by jdef** [comment]
@dchen1107 agreed. the `ok` stuff made more sense when the iteration loop had more junk in it. better now.

**Comment by dchen1107** [comment]
LGTM

**Pull request merged**